tests/services/test_record_service_files.py

```diff
@@ -390,3 +393,57 @@ def test_fail_to_add_files_to_draft_with_files_disabled(
     with pytest.raises(InvalidOperationError):
         add_file_to_draft(
             service.draft_files, draft.id, "file.txt", identity_simple)
+
+
+def test_hard_delete_soft_deleted(
+        app, service, identity_simple, input_data):
+    draft = service.create(identity_simple, input_data)
+    add_file_to_draft(service.draft_files, draft.id, 'test', identity_simple)
+    record = service.publish(draft.id, identity_simple)
+    draft_model = service.draft_cls.model_cls
+
+    assert len(draft_model.query.filter(
+        draft_model.is_deleted == True  # noqa
+    ).all()) == 1
+    Draft.cleanup_drafts(timedelta(seconds=0))
+    db.session.commit()
+
+    assert len(draft_model.query.filter(
+        draft_model.is_deleted == True  # noqa
+    ).all()) == 0
+
+
+def test_hard_delete_soft_deleted_task(
```

Comment:
Should `task` related tests go to a different file, e.g. `test_service_tasks.py`? to be more independent?

i.e. here testing service level functions (like above), then testing task level in a specific file...